src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -2112,9 +2112,9 @@ subroutine gsi_fv3ncdf2d_read(fv3filenamegin,it,ges_z,ges_t2m,ges_q2m,ges_howv)
              k=k_tsea
           else if( trim(name)=='sheleg'.or.trim(name)=='SHELEG' ) then
              k=k_snwdph
-          else if( trim(name)=='stc'.or.trim(name)=='STC' ) then
+          else if( trim(name)=='stc'.or.trim(name)=='tslb' ) then
              k=k_stc 
-          else if( trim(name)=='smc'.or.trim(name)=='SMC' ) then
+          else if( trim(name)=='smc'.or.trim(name)=='smois' ) then
```

Comment:
@xyzemc  could we also add a certain exception handling for this part . For example , an array has_sfc_var(n2d), when the variable is found, the correspondg has_sfc_var(k)=1.    and, when, for example, has_sfc_var(k_stc) = 1 , sfcn2d(:,:,k_stc) would be assigned to the corresponding variable, like temp_soil, otherwise, the later's default value would be used (which is initialized as zero), or let GSI abort if some variables are missing and there is no non-aborting treatment available.

Reply:
logical variable 'sfc_var_exist' has been added into the code to handle the surface variable values when any of the variable is not existing in the restart file. This will avoid the undefined value.

Reply:
@xyzemc  since sfc_var_exist is declared as logical,  could they directly be assigned ".true." or ".false".   I know =1 (or non negative integers) would, typically, let the logical varible to be true . But I am not sure if it  is the standard and direct assignment using ".true." or ".false" would have better readability (for example, I would not take it as integer ).

Reply:
> @xyzemc since sfc_var_exist is declared as logical, could they directly be assigned ".true." or ".false". I know =1 (or non negative integers) would, typically, let the logical varible to be true . But I am not sure if it is the standard and direct assignment using ".true." or ".false" would have better readability (for example, I would not take sfc_var_exist as integer ).

@xyzemc  What is your opinion on this comment?

Reply:
I agree with you we should follow the strandard of GSI. I will change it to '.true.' and '.false.'